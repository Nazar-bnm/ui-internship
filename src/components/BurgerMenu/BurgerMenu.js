import React from 'react';
import './BurgerMenu.scss';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const BurgerMenu = ({ menuItemsList }) => {
  const renderMenu = () => menuItemsList.map(({ itemName }) => {
    return (
      <Link key={itemName} to={'/404'} className="burger-menu-link">
        <li key={itemName} className="burger-list-item">{ itemName }</li>
      </Link>
    );
  });

  return (
    <div className="burger-menu-container">
      <nav className="burger-menu-navigation-container">
        <input type="checkbox" id="burgerbutton" />
        <label htmlFor="burgerbutton" className="burgerimage" title="Navigation">&#x2630;</label>
        <ul className="burger-list">
          {renderMenu()}
        </ul>
      </nav>
    </div>
  );
};

BurgerMenu.propTypes = {
  menuItemsList: PropTypes.array,
};

export default BurgerMenu;
