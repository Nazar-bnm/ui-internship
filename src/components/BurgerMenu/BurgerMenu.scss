@import '../../scss/global_styles';

.burger-menu {
  display: flex;
  width: 50px;
  padding-right: 10px;

  @keyframes open {
    from {
      left: -80vw;
    }
    to {
      left: 0;
    }
  }

  .icon {
    font-size: 1.7rem;
  }

  #burger-menu__button {
    display: none;

    &:checked~.burger-menu__list {
      @include flex();
      animation-name: open;
      animation-duration: 0.8s;
    }

    &:checked~.burger-menu__image {
      position: absolute;
      cursor: pointer;
      transition: transform 0.8s ease-in; 
      transform: rotate(360deg);
    }
  }

  &__image {
    margin-top: -10px;
    margin-left: 15px;
    font-size: 1.7rem;
    cursor: pointer;
    transition: transform 0.8s ease; 
  }

  &__list {
    display: none;
    position: relative;
    top: 30px;
    padding: 0;
    list-style: none;
  }

  &__navigation {
    display: flex;
    height: 30px;
    z-index: 100;
  }

  &__list-item {
    display: block;
    width: 120px;
  }

  &__link {
    font-size: 0.9rem;
    text-decoration: none;
    background-color: $white;
    color: $black;
    display: block;
    padding: 15px;
    height: 50px;
    transition: background 0.5s ease;
    box-sizing: border-box;
    border-bottom: 1px solid $alto;

    &:hover {
      background-color: $alto;
      color: $mine-shaft;
    }
  }
}
