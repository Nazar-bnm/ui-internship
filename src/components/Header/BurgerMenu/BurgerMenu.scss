@keyframes open {
  from {
      left: -100vw;
  }
  to {
      left: 0;
  }
}


#burgerbutton {
  display: none;
}

/* You probably want to replace the text part of the label element with a background via css in productive use */

.burgerimage {
  margin: -10px 0 10px 10px;
  font-size: 2.7rem;
  cursor: pointer;
  transition: transform 0.8s ease; 
}

.burger-list {
  display: none;
  margin-left: 5px;
  list-style: none;
}

/* Example styling of the menu */

nav {
  display: flex;
  height: 30px;
  z-index: 100;
}

nav ul {
  display: none;
  position: relative;
  top: 35px;
  margin: 0;
  padding: 0;
  width: 50vw;
  list-style: none;
}

nav ul li {
  display: block;
  width: auto;
}

nav ul li a {
  text-decoration: none;
  // color: #ff0000;
  background-color: #ffffff;
  // border-bottom: 1px solid #8b0000;
  display: block;
  padding: 15px;
  height: 50px;
  transition: background 0.5s ease;
  box-sizing: border-box;
}

nav ul li a:hover {
  background-color: #ffc0cb;
  color: #8b0000;
}

/* Displaying the menu when #burgerbutton is checked */

#burgerbutton:checked~ul {
  display: flex;
  flex-direction: column;
  animation-name: open;
  animation-duration: 0.8s;
}

#burgerbutton:checked~.burgerimage {
  position: absolute;
  left: 10px;
  font-size: 3rem;
  cursor: pointer;
  transition: transform 0.8s ease; 
}

/* And rotate the visible burger button label when its activated */

#burgerbutton:checked~.burgerimage {
  transform: rotate(360deg);
}

@media(min-width:720px) {
  nav ul {
      display: block;
      position: static;
  }
  nav ul li {
      display: inline-block;
  }
  nav ul li a {
      border: none;
  }
  .burgerimage {
      display: none;
  }
}