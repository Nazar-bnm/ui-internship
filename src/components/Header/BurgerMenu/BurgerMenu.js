import React from 'react';
import './BurgerMenu.scss';

const BurgerMenu = () => {
  return (
    <div className="burger-menu-container">
      <nav>
        <input type="checkbox" id="burgerbutton" />
        <label htmlFor="burgerbutton" className="burgerimage" title="Navigation">&#x2630;</label>
        <ul className="burger-list">
          <li>
            <a className="burger-menu-link" href="#" title="Link">Link</a>
          </li>
          <li>
            <a className="burger-menu-link" href="#" title="Link">Link</a>
          </li>
          <li>
            <a className="burger-menu-link" href="#" title="Link">Link</a>
          </li>
          <li>
            <a className="burger-menu-link" href="#" title="Link">Link</a>
          </li>
          <li>
            <a className="burger-menu-link" href="#" title="Link">Link</a>
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default BurgerMenu;
