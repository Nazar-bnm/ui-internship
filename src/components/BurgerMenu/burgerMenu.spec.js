import React from 'react';
import { shallow } from 'enzyme';
import BurgerMenu from './BurgerMenu';

describe('<BurgerMenu />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      menuItemsList: [
        { itemName: 'woman' },
        { itemName: 'man' },
        { itemName: 'kids' },
        { itemName: 'accessories' },
        { itemName: 'what\'s new' },
        { itemName: 'brands' },
        { itemName: 'sale' },
        { itemName: 'blog' },
      ],
    };

    wrapper = shallow(
      <BurgerMenu {...props} />
    );
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
