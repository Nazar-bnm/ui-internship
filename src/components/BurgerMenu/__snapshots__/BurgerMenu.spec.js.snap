// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurgerMenu /> should match snapshot 1`] = `
<div
  className="burger-menu-container"
>
  <nav
    className="burger-menu-navigation-container"
  >
    <input
      id="burgerbutton"
      type="checkbox"
    />
    <label
      className="burgerimage"
      htmlFor="burgerbutton"
      title="Navigation"
    >
      â˜°
    </label>
    <ul
      className="burger-list"
    >
      <Link
        className="burger-menu-link"
        key="woman"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="woman"
        >
          woman
        </li>
      </Link>
      <Link
        className="burger-menu-link"
        key="man"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="man"
        >
          man
        </li>
      </Link>
      <Link
        className="burger-menu-link"
        key="kids"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="kids"
        >
          kids
        </li>
      </Link>
      <Link
        className="burger-menu-link"
        key="accessories"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="accessories"
        >
          accessories
        </li>
      </Link>
      <Link
        className="burger-menu-link"
        key="what's new"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="what's new"
        >
          what's new
        </li>
      </Link>
      <Link
        className="burger-menu-link"
        key="brands"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="brands"
        >
          brands
        </li>
      </Link>
      <Link
        className="burger-menu-link"
        key="sale"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="sale"
        >
          sale
        </li>
      </Link>
      <Link
        className="burger-menu-link"
        key="blog"
        to="/404"
      >
        <li
          className="burger-list-item"
          key="blog"
        >
          blog
        </li>
      </Link>
    </ul>
  </nav>
</div>
`;
